www/src/components/url-query.js

```diff
@@ -0,0 +1,50 @@
+import { Component } from "react"
+import { withRouter } from "react-router-dom"
+import qs from "qs"
+
+class URLQuery extends Component {
+  state = {}
+
+  componentDidMount = () => {
+    const {
+      history,
+      location: { search },
+    } = this.props
+
+    this.getDerivedStateFromQuery(search)
```

Comment:
too much? :joy:

Reply:
nice... i use a [react-router-state-manager](https://github.com/sw-yx/gatsby/blob/starterShowcase2/www/src/utils/react-router-state-manager.js) utility if you wanna just use that